modules/fedimint-mint-client/src/oob.rs

```diff
@@ -44,11 +44,18 @@ pub enum MintOOBStates {
     /// recipient to reissue it or the user to trigger a refund.
     CreatedMulti(MintOOBStatesCreatedMulti),
     /// The user has triggered a refund.
+    UserRefundMulti(MintOOBStatesUserRefundMulti),
+    /// The user has triggered a refund (legacy).
     UserRefund(MintOOBStatesUserRefund),
     /// The timeout of this out-of-band transaction was hit and we attempted to
     /// refund. This refund *failing* is the expected behavior since the
     /// recipient is supposed to have already reissued it.
     TimeoutRefund(MintOOBStatesTimeoutRefund),
+    /// Successfully refunded in a single tx after `UserRefundMulti`
+    RefundedMulti,
+    /// Refunding in a single tx after `UserRefundMulti` failed, started bunch
+    /// of refunds per every spendable note
+    RefundedMultiFallback(MintOOBStatesRefundedMultiFallback),
 }
```

Comment:
Same here: either only add states at the end or maybe just get rid of the old variant (preferable imo, less code).

Also, we might be getting to the point where we want manual annotation of enum variant indices and should adapt the `Encodable` and `Decodable` macros accordingly.

Reply:
The old state doesn't have all the info (list of notes) to upgrade it to the new state. @elsirion